pkgs/tools/system/osquery/default.nix

```diff
@@ -1,59 +1,68 @@
 { stdenv, lib, fetchFromGitHub, pkgconfig, cmake, pythonPackages
 , udev, audit, aws-sdk-cpp, cryptsetup, lvm2, libgcrypt, libarchive
-, libgpgerror, libuuid, iptables, apt, dpkg, lzma, lz4, bzip2, rpm
+, libgpgerror, libuuid, iptables, dpkg, lzma, bzip2, rpm
 , beecrypt, augeas, libxml2, sleuthkit, yara, lldpd, google-gflags
-, thrift, boost, rocksdb_lite, cpp-netlib, glog, gbenchmark, snappy
-, openssl, linenoise-ng, file, doxygen, devicemapper
-, gtest, sqlite
+, thrift, boost, rocksdb_lite, glog, gbenchmark, snappy
+, openssl, file, doxygen, fetchpatch
+, gtest, sqlite, fpm, zstd, rdkafka, rapidjson, path
 }:
 
 let
+
   thirdparty = fetchFromGitHub {
     owner = "osquery";
     repo = "third-party";
-    rev = "6919841175b2c9cb2dee8986e0cfe49191ecb868";
-    sha256 = "1kjxrky586jd1b2z1vs9cm7x1dxw51cizpys9kddiarapc2ih65j";
+    rev = "4ef099c31a1165c5e7e3a699f9e4b3eb68c3c3d9";
+    sha256 = "1vm0prw4dix0m51vkw9z0vwfd8698gqjw499q8h604hs1rvn6132";
   };
 
 in
 
 stdenv.mkDerivation rec {
   name = "osquery-${version}";
-  version = "2.5.2";
+  version = "3.2.2";
 
   # this is what `osquery --help` will show as the version.
   OSQUERY_BUILD_VERSION = version;
+  OSQUERY_PLATFORM = "nixos;${builtins.readFile "${builtins.toString path}/.version"}";
 
   src = fetchFromGitHub {
     owner = "facebook";
     repo = "osquery";
     rev = version;
-    sha256 = "16isplk66qpvhrf041l0lxb4z6k7wwd1sg7kpsw2q6kivkxpnk3z";
+    sha256 = "0qwj4cy6m25sqwb0irqfqinipx50l4imnz1gqxx147vzfwb52jlq";
   };
 
-  patches = [ ./misc.patch ] ++ lib.optional stdenv.isLinux ./platform-nixos.patch;
+  patches = [
+    ./misc.patch
+    (fetchpatch {
+      url = "https://patch-diff.githubusercontent.com/raw/facebook/osquery/pull/4267.patch";
```

Comment:
This url will break if changes are made to the pull request. @flokli could you make a dedicated branch for this that is independent from the pull request and that will not change?